cmd/bootstrap/cmd.go

```diff
@@ -0,0 +1,275 @@
+package bootstrap
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/cloudformation"
+	cfTypes "github.com/aws/aws-sdk-go-v2/service/cloudformation/types"
+	common "github.com/openshift-online/ocm-common/pkg/aws/validations"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+
+	"github.com/openshift/rosa/pkg/aws/tags"
+	"github.com/openshift/rosa/pkg/ocm"
+	"github.com/openshift/rosa/pkg/rosa"
+)
+
+func NewRosaBootstrapCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "bootstrap",
+		Short: "Bootstrap quick start vpc",
+		Long:  "Bootstrap quick start vpc. This vpc can be used to create a HCP cluster",
+		Example: `  # Create a rosa-quickstart-vpc
+  rosa bootstrap rosa-quickstart-default-vpc --param Name=rosa-quickstart-default --param VpcCidr=10.0.0.0/16 ` +
+			`--param Region=us-west-2 --param AvailabilityZonesCount=2 --param Tags=Key1=Value1`,
+		Args:   cobra.ExactArgs(1),
+		Hidden: true,
+		Run:    run,
+	}
+	cmd.Flags().StringArrayVar(&args.params, "param", []string{}, "List of parameters")
+	return cmd
+}
+
+var args struct {
+	params []string
+}
+
+func run(cmd *cobra.Command, arg []string) {
+	r := rosa.NewRuntime().WithAWS().WithOCM()
+	defer r.Cleanup()
+
+	orgID, _, err := r.OCMClient.GetCurrentOrganization()
+	if err != nil {
+		r.Reporter.Errorf(err.Error())
+		os.Exit(1)
+	}
+
+	parsedParams, parsedTags := parseParams(args.params)
+
+	// Extract the first non-`--param` argument to use as the template command
+	var templateCommand string
+	for _, arg := range arg {
+		if !strings.HasPrefix(arg, "--param") {
+			templateCommand = arg
+			break
+		}
+	}
+
+	templateFile := selectTemplate(templateCommand)
+	if templateFile == "" {
+		r.Reporter.Errorf("No suitable template found for the specified parameters")
+		os.Exit(1)
+	}
+
+	r.OCMClient.LogEvent("RosaQuickStartVpc",
+		map[string]string{
+			ocm.Account:      r.Creator.AccountID,
+			ocm.Organization: orgID,
+			"template":       templateFile,
+		},
+	)
+
+	err = createVPCStack(templateFile, parsedParams, parsedTags)
+	if err != nil {
+		r.Reporter.Errorf(err.Error())
+		os.Exit(1)
+	}
+
+	r.Reporter.Infof("VPC created")
+}
+
+// parseParams converts the list of parameter strings into a map and sets default values
+func parseParams(params []string) (map[string]string, map[string]string) {
+	result := map[string]string{
```

Comment:
@den-rgb These defaults should probably be stored in the CloudFormation template. Take a look here: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/parameters-section-structure.html

Another move to keep this command generic and have the CloudFormation Template control most things.